tests/jballerina-unit-test/src/test/resources/test-src/types/regexp/regexp_type_test.bal

```diff
@@ -105,6 +106,22 @@ function testRegExpWithUserDefinedType() {
     assertEquality(re `AB*|[^c-d]{1,5}`, x7);
 }
 
+type T1 string:RegExp & readonly;
+type T2 regexp:RegExp & readonly;
+
+type Foo record {|
+    int e;
+    readonly regexp:RegExp f;
+|};
+
+Foo & readonly rf = {e: 1, f: re `test`};
+
+function testRegExpReadonlyLocalVars() {
+    string:RegExp & readonly _ = re `test`;
+    T1 & readonly _ = re `test`;
+    (T2 & readonly) & string:RegExp _ = re `test`;
+}
+
```

Comment:
Let's add some negative tests too. E.g., trying to use it against other immutable types that are incompatible.

Probably also something like

```
(stream<int>|regexp:RegExp) & readonly x = new;
```

Reply:
Added in [0ae6e3](https://github.com/ballerina-platform/ballerina-lang/pull/41843/commits/0ae6e361aead9db6235956d2a6dd141a62ca8396).